geps/gep-713/index.md

```diff
@@ -3,1155 +3,1114 @@
+  end
+```
 
-In all of these cases, we should aim to keep the troubleshooting distance low;
-that is, that there should be a minimum of hops required between objects from the
-one owned by the user to the one responsible for a setting.
+In the example above, the expected outcome of the process is:
 
-Another way to think of the troubleshooting distance in this context is "How many
-`kubectl` commands would the user need to do to understand that a Policy is relevant,
-which Policy is relevant, and what configuration the full set of Policy is setting?"
+* `c1` is augmented by `m1`, whenever activated in the context of `b1`;
+* `c1` is augmented by the combination of `m1` + `m2`, whenever activated in the context of `b2`;
+* `c2` is augmented by the combination of `m1` + `m2`.
 
-##### Application Developer Discoverability
+In the most trivial case where metaresources can only directly target the objects whose behavior they intend to augment (i.e. instances of `C` without any indirections) and no metaresource specs are merged at all, the outcome of the process of calculating effective metaresources is simplified to a 1:1 mapping between metaresource and target object at most, where the declared metaresource equals the effective one, with no combinatorial specs nor contextual variations.
 
-How does Ana, or any Application Developer who owns one or more Route objects know
-that their object is affected by Policy, which Policy is affecting it, and what
-the content of the Policy is?
+### Classes of metaresources
 
-The best outcome is that Ana needs to look only at a specific route to know what
-Policy settings are being applied to that Route, and where they come from.
-However, some of the other problems below make it very difficult to achieve this.
+The following two classes of metaresource kinds are defined: *Direct* and *Inherited*.
 
-##### Policy Admin Discoverability
+#### Direct
 
-How does the Policy Admin know what Policy is applied where, and what the content
-of that Policy is?
-How do they validate that Policy is being used in ways acceptable to their organization?
-For any given Policy object, how do they know how many places it's being used?
+* A single kind supported in `spec.targetRefs.kind`
+* Effects of the metaresources do not span across the hierarchy, i.e. the _Declared target kind_ is equal to the _Effective target kind_
+* *None* is the only merge strategy supported
+* If supported, could typically be implemented by directly extending the API of the target kind with the fields otherwise defined at the metaresource (e.g. Gateway API filter)
 
-##### Cluster Admin Discoverability
+#### Inherited
 
-The Cluster Admin has similar concerns to the Policy Admin, but with a focus on
-being able to determine what's relevant when something is broken.
+* Superset of the above
+* Any metaresource kind that do not comply with at least one characteristic of the Direct class of metaresources
```

Comment:
These are quite ambiguous without examples. Could you first define why "meta-resource classes" are necessary and why Direct and Inherited are relevant?

Reply:
Mainly to ease the transition from the current state of the GEP/s.

GEP-713 was once forked into two GEPs – 2648 and 2649, respectively for Direct and Inherited Policies. We're now merging them back into one single GEP.

The difference between these two classes (Direct and Inherited), if relevant, should emerge from the properties of metaresources specified, including and especially from the concept of merge strategy.

Reply:
Those 2 GEPs are marked as declined. So this GEP should reintroduce those concepts before including them in the spec.

Reply:
Sorry. IDK if I follow. There's no spec other than the GEP itself. What do you mean by the "GEP should reintroduce before including in the spec"? This subsection of the guide-level explanation is meant to introduce/define the concepts.

Reply:
I meant we should describe the motivation for "meta-resource classes` as it pertains to this GEP instead of assuming it is a well understood concept.

Reply:
Done